File: public/docs/_examples/component-styles/dart/lib/hero_app_main_component.dart
import 'package:angular2/core.dart';

import 'hero.dart';
import 'hero_details_component.dart';
import 'hero_controls_component.dart';
import 'quest_summary_component.dart';

@Component(
    selector: 'hero-app-main',
    template: '''
      <quest-summary></quest-summary>
      <hero-details [hero]="hero" [class.active]="hero.active">
        <hero-controls [hero]="hero"></hero-controls>
      </hero-details>''',
    directives: const [
      HeroDetailsComponent,
      HeroControlsComponent,
      QuestSummaryComponent
    ])
class HeroAppMainComponent {
  @Input() Hero hero;
}

File: public/docs/_examples/component-styles/ts/app/hero-app-main.component.ts
import { Component, Input } from '@angular/core';

import { Hero } from './hero';
import { HeroDetailsComponent } from './hero-details.component';
import { HeroControlsComponent } from './hero-controls.component';
import { QuestSummaryComponent } from './quest-summary.component';

@Component({
  selector: 'hero-app-main',
  template: `
    <quest-summary></quest-summary>
    <hero-details [hero]=hero [class.active]=hero.active>
      <hero-controls [hero]=hero></hero-controls>
    </hero-details>
  `,
  directives: [HeroDetailsComponent, HeroControlsComponent, QuestSummaryComponent]
})
export class HeroAppMainComponent {
  @Input() hero: Hero;
}
